test/functional/test_framework/test_framework.py

```diff
@@ -596,24 +596,24 @@ def connect_nodes(self, a, b):
         self.wait_until(lambda: sum(peer['bytesrecv_per_msg'].pop('verack', 0) == 24 for peer in to_connection.getpeerinfo()) == to_num_peers)
 
     def disconnect_nodes(self, a, b):
-        def disconnect_nodes_helper(from_connection, node_num):
-            def get_peer_ids():
+        def disconnect_nodes_helper(node_a, node_b):
+            def get_peer_ids(from_connection, node_num):
```

Comment:
nit: Since you're touching, I think adding type hints here would be helpful. Also, for consistency would prefer calling it `node` instead of `connection` and `index` instead of `num`?
```suggestion
    def disconnect_nodes(self, a: int, b: int) -> None:
        def disconnect_nodes_helper(node_a: TestNode, node_b: TestNode) -> None:
            def get_peer_ids(from_node: TestNode, to_node_index: int) -> List[int]:
```

Reply:
I kept the name as-is to make the diff minimal. If there are additional changes, they can be made in a follow-up.

If there is a follow-up, it would probably be fine to also make `to_node_index` of type `TestNode`.

Reply:
@stickies-v maybe you can follow up here?